Dockerfile

```diff
@@ -1,5 +1,27 @@
 FROM redash/base:latest
 
+ENV DEBIAN_FRONTEND=noninteractive \
+    MOZ_HEADLESS=1 \
+    FIREFOX_VERSION=59.0 \
+    GECKODRIVER_VERSION=0.20.0
+
+# Install requirements for UI tests
+RUN dependencies=' \
+        bzip2 \
+        curl \
+        firefox \
+    ' \
+    && set -x \
+    && apt-get -qq update && apt-get -qq install --no-install-recommends -y $dependencies \
+    && apt-get -y purge firefox \
+    && rm -rf /var/lib/apt/lists/* /var/cache/apt/*
+
+COPY bin/install-firefox.sh /usr/local/bin/install-firefox
+COPY bin/install-geckodriver.sh /usr/local/bin/install-geckodriver
```

Comment:
To reduce the number changes to the orginal Dockerfile, let's just use the `COPY` to /app and call the installation scripts from there.

Reply:
We can speed up Docker builds by installing Firefox and Geckodriver in an extra step and making use of Docker's caching feature. Otherwise every code change will require a new Docker build which then downloads Firefox and Geckodriver.